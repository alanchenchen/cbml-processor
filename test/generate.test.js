const { readFile } = require('fs/promises')
const { resolve } = require('path')
const { parseCBMLEle } = require('../lib/parser')
const { generateCode } = require('../lib/generater')

let CBMLRootEle

beforeAll(async () => {
  const demoPath = resolve(__dirname, './parse.demo.js')
  const code = await readFile(demoPath, { encoding: 'utf-8' })
  CBMLRootEle = parseCBMLEle(code)
})

describe('test CBML generater', () => {
  test('generate CBML Snapshot', () => {
    expect(generateCode(CBMLRootEle)).toMatchSnapshot()
  })
})