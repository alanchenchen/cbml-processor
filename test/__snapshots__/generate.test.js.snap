// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CBML generater generate CBML Snapshot 1`] = `
"/*<visible lib='golang'>
alert('user golang');
</visible>*/
const a = 1
/*<do />*/

/*<block>*/
alert('user golang');
const a = 1
/*</block>*/"
`;
